src/server/game/Movement/MovementGenerators/TargetedMovementGenerator.cpp

```diff
@@ -168,8 +168,15 @@ bool TargetedMovementGeneratorMedium<T, D>::DoUpdate(T* owner, uint32 time_diff)
     if (i_recheckDistance.Passed())
     {
         i_recheckDistance.Reset(100);
+
         //More distance let have better performance, less distance let have more sensitive reaction at target move.
-        float allowed_dist = owner->GetCombatReach() + sWorld->getRate(RATE_TARGET_POS_RECALCULATION_RANGE);
+        float allowed_dist = 0.0f;
+
+        if ((owner->GetTypeId() == TYPEID_UNIT && owner->IsPet()) && (owner->GetCharmerOrOwnerGUID() == i_target->GetGUID()))
```

Comment:
> are you sure == TYPEID_UNIT check is required in addition to IsPet() ?

Probably not anymore. The original IsPet() was this:

```C++
bool IsPet() const { return m_unitTypeMask & UNIT_MASK_PET; }
```

and I used the TYPEID_UNIT check just to make sure that owner was in fact a unit. The new IsPet() is quite different though:

```C++
bool IsPet() const { return GetHigh() == HighGuid::Pet; }
```

HighGuid is something that was added after my original commit.